iree/compiler/Conversion/LinalgToSPIRV/LinalgTileAndFusePass.cpp

```diff
@@ -547,12 +624,29 @@ void LinalgTileAndFusePass::runOnOperation() {
         OwningRewritePatternList vectorUnrollPatterns;
         populateVectorUnrollPatterns(context, vectorUnrollPatterns);
         applyPatternsAndFoldGreedily(funcOp, std::move(vectorUnrollPatterns));
+
+        OwningRewritePatternList canonicalizationPatterns;
+        vector::populateVectorSlicesLoweringPatterns(canonicalizationPatterns,
+                                                     context);
+        applyPatternsAndFoldGreedily(funcOp,
+                                     std::move(canonicalizationPatterns));
         LLVM_DEBUG({
           llvm::dbgs() << "--- After Vector Unroll ---\n";
           funcOp.print(llvm::dbgs(), OpPrintingFlags().useLocalScope());
           llvm::dbgs() << "\n\n";
         });
       }
+
+      {
+        linalg::hoistViewAllocOps(funcOp);
```

Comment:
Is this to move the alloc for workgroup memory to the start of the function?

I am planning to change how this works, and adapt the allocation function callback to just inesrt the allocation at the start of the function. Also making the deallocation function callback a no-op. That will match semantics of workgroup memory better. With that this shouldnt be needed right?

Reply:
Sure. Please add a TODO here and assign to me?

Reply:
Added a TODO for this.